metrics-core/src/main/java/com/yammer/metrics/util/OptionalTimer.java

```diff
@@ -0,0 +1,45 @@
+package com.yammer.metrics.util;
+
+import java.io.Closeable;
+import java.util.Optional;
+
+import com.yammer.metrics.core.Timer;
+
+public class OptionalTimer {
+  private static final OptionalTimer EMPTY = new OptionalTimer();
+  private final Optional<Timer> timer;
+
+  private OptionalTimer(Timer timer) {
+    this.timer = Optional.of(timer);
+  }
+
+  private OptionalTimer() {
+    this.timer = Optional.empty();
+  }
+
+  public Closeable time() {
+    if (timer.isPresent()) {
```

Comment:
nit:
```java
return timer.map(Timer::time).orElse(NullCloseable.INSTANCE);
```

Reply:
I did this but I had to put a type annotation on `map`.